### ISSUE TITLE
Everything AWS as peer dependency?

### ISSUE THREAD

**Issue by pocesar** [punchcard/punchcard#59]
CDK released a new 1.19 version, and installing the new version, because of their need for 1.18 version of peer dependencies, gives the expected warnings about versions that could not be found:

![image](https://user-images.githubusercontent.com/461084/71019487-25a84c00-20d9-11ea-9744-14396f87a07e.png)

I know dealing with those high iteration libraries, like aws-sdk itself, can be frustrating, but if they use semver strictly (which I think they do), it would be safe to rely on "peerDependencies" instead of "dependencies" and use "^1.0.0", but at the same time make sure there are no regressions (like testing for node 10, 12 and 13 at the same time using test matrix)

**Comment by sam-goodwin** [comment]
I have a PR up for fixing the versions: https://github.com/punchcard/punchcard/pull/60. Does it look right to you?

**Comment by sam-goodwin** [comment]
I've unfortunately had to release a new version of Punchcard every time the CDK does and am not sure how to fix it.

**Comment by sam-goodwin** [comment]
Maybe I need to take an explicit dependency on every cdk package? That way there are no nested CDK dependencies in the `node_modules`.

**Comment by pocesar** [comment]
yeah, since CDK is choosing the path to not use carrets in version fields, that should be safer... wish they were using semver strictly and minor and patches didn't mean breaking changes at all, since they are "1.0"

**Comment by pocesar** [comment]
@sam-goodwin maybe a @dependabot would make it easier to test it before hand as a PR, without you having to be "on alert" for new versions

**Comment by sam-goodwin** [comment]
dependabot is awesome!
